### DISCUSSION TITLE
Delete users or items?

### DISCUSSION THREAD

**Discussion by mimosel** [leihs/leihs#1517] [Q&A / Community Support] [answered]
Hello all. First of all, we don't use leihs yet, but we are currently researching which software solution could support us for our equipment rental. We want to manage about 250 devices and have potentially >20000 users.

I have made a few phone calls and asked one or the other university what their experiences with leihs are. I was told independently by two people that you can't delete devices/users if they have ever been involved in a borrowing process. Is that the case?

If so, I would find that problematic. We often have devices break, which are then sometimes replaced by devices with similar functionality but from a different manufacturer - in other words, by other devices, so I would have to delete the original device. Our users are students who usually leave the university after a few years and are then no longer eligible to borrow - I would also have to be able to delete them, regardless of whether they had once borrowed something or not.

What are your experiences in this regard? Is it possible to delete devices/users without any problems?

**Comment by winampdevil** [comment]
Hello mimosel,
my experience: 
you can delete users in 2 ways:
direct delete, if the user wasn't involved in any borrow-process.
or deletion with transfer of all the borrow-data to another user. This is a step within the deletion-process. We use a dummy-user for that.

Deleting things is not really possible, but you can mark them as non-borrowable, broken or retired.
You can edit your inventory, maybe that's a solution for you.

Hope that helps
Frank

**Comment by neoheat** [comment, marked as answer]
Hello @mimosel
We will be happy to answer your questions about Leihs
 
# Deletion of items
From experience at ZHdK lending, instead of deleting items, they are retired and replaced with new items. The reason is that you may still want to trace the history of the old item. The new item can then be lent out again as a replacement.
 
# Deleting users
Leihs does not delete users directly from the system. However, there is a way to anonymize users during user sync with AIM (e.g. Azure AD) using an appropriate script. The user itself remains in the system (only with UUID), in order not to lose the relations to possible lendings or contracts.
